src/receive_imf.rs

```diff
@@ -1562,7 +1564,7 @@ INSERT INTO msgs
 ON CONFLICT (id) DO UPDATE
 SET rfc724_mid=excluded.rfc724_mid, chat_id=excluded.chat_id,
     from_id=excluded.from_id, to_id=excluded.to_id, timestamp_sent=excluded.timestamp_sent,
-    type=excluded.type, msgrmsg=excluded.msgrmsg,
+    type=excluded.type, state=min(state,13), msgrmsg=excluded.msgrmsg,
```

Comment:
Doesn't this mean that if an outgoing message isn't fully downloaded (with multi-device) and then the user clicks on "Download", the message state will change from OutDelivered/OutMdnRcvd to InNoticed?

Reply:
Of course it should be `max(new_state,13)`, not just 13. Fixed it. One note: `state` var in the code doesn't always contain the actual message state (unlike in the db), but this may affect only tests and currently there are no such ones.